memgpt/memory.py

```diff
@@ -310,24 +360,31 @@ def __init__(self, agent_config, top_k: Optional[int] = 100):
 
         self.top_k = top_k
         self.agent_config = agent_config
-        config = MemGPTConfig.load()
+        self.config = MemGPTConfig.load()
 
         # create embedding model
         self.embed_model = embedding_model()
-        self.embedding_chunk_size = config.embedding_chunk_size
+        self.embedding_chunk_size = self.config.embedding_chunk_size

```

Comment:
Not sure how to handle these properties in the case where a user swaps embedding models with an agent. Create a new archival memory object? Or make the `EmbeddingArchivalMemory` class be `embedding_model` agnostic, and that instead is held in the record / `Passage` metadata?